pip/req/req_install.py

```diff
@@ -1157,6 +1157,11 @@ def parse_editable(editable_req, default_vcs=None):
 
     if '+' not in url:
         if default_vcs:
+            warnings.warn(
+                "--default-vcs has been deprecated and will be removed in "
+                "the future.",
+                RemovedInPip10Warning,
```

Comment:
It could be pip 11, it's just I want to get rid of it ;)

This option seems really useless to me (the only usecase I see is to put it in your pip.conf to be able to `pip install -e https://github.com/pypa/pip.git` instead of `pip install -e git+https://github.com/pypa/pip.git`)

We can also start with `RemovedInPip10Warning` and if people complain, bump it to `RemovedInPip11Warning` in pip 9.1:grin: